app/uk/gov/hmrc/transitmovements/models/MovementType.scala

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovements.models
+
+case class MovementType(value: String) extends AnyVal
```

Comment:
As we only have two movement types, "arrival" and "departure", we should make this a sealed abstract class with two case objects, something like:

```scala
sealed abstract class MovementType(val value: String)

object MovementTypes {
    case object Arrival extends MovementType("arrival")
    case object Departure extends MovementType("departure")
}
```

and use these objects, rather than wrapping the strings in multiple places (that way, we're not repeating the magic string and if we need to refactor for whatever reason, it'll be easier to do - plus it avoids typos!).

Reply:
updated the code, please re-review